### PR TITLE
Save on AdHoc Question Title Click

### PR DESCRIPTION
**Pull request by npfitz**
Epic #22826 

Adds a click handler to the title of an unsaved question that will open the Save Question Modal, as well as re-locates the save button and removes the icon in the subhead data source.

image

### CODE REVIEW

**Review by iethree — CHANGES_REQUESTED** [review]
Works and looks great, I think there may have been an unintended CSS border added though.

File: frontend/src/metabase/query_builder/components/view/ViewHeader.styled.jsx [hunk 1]
```diff
@@ -45,6 +41,9 @@ export const SaveButton = styled(Link)`
   border-radius: 8px;
   background-color: ${color("bg-white")};
 
+  margin-left: 0.5rem;
+  border-left: 1px solid ${color("border")};
+
```
[the remark this answers is not in the document]
Comment (npfitz) [reply frontend/src/metabase/query_builder/components/view/ViewHeader.styled.jsx:46]: I did (in the query builder header). Alternatively I could set up a container to put the button in that has the border/margin

File: frontend/src/metabase/query_builder/components/view/QuestionDescription.jsx [hunk 2]
```diff
@@ -39,14 +42,14 @@ const QuestionDescription = ({
             breakouts.length,
           )
         : breakouts.map(breakout => breakout.displayName()).join(t` and `);
-    if (aggregationDescription && breakoutDescription) {
+    if (aggregationDescription || breakoutDescription) {
       return (
-        <span>{t`${aggregationDescription} by ${breakoutDescription}`}</span>
+        <StyledSpan onClick={onClick}>
+          {[aggregationDescription, breakoutDescription]
+            .filter(x => !!x)
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/view/QuestionDescription.jsx:49]: nit: `filter(Boolean)`

File: frontend/src/metabase/query_builder/components/view/QuestionDescription.styled.tsx [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import styled from "@emotion/styled";
+
+export const StyledSpan = styled.span`
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/view/QuestionDescription.styled.tsx:3]: nit: `AggregationAndBreakoutDescription`?

**Review by iethree — APPROVED** [review]
🎆

**Pull request merged**